Gems/Atom/Feature/Common/Code/Source/Silhouette/SilhouetteFeatureProcessor.cpp

```diff
@@ -185,7 +167,35 @@ namespace AZ::Render
 
         if (auto pass = RPI::PassSystemInterface::Get()->CreatePassFromRequest(&compositePassRequest); pass != nullptr)
         {
+            m_compositePass = pass.get();
             renderPipeline->AddPassAfter(pass, postProcessPassName);
         }
     }
+
+    void SilhouetteFeatureProcessor::OnRenderPipelineChanged(AZ::RPI::RenderPipeline* pipeline, [[maybe_unused]] AZ::RPI::SceneNotification::RenderPipelineChangeType changeType)
+    {
+        // update our render pass members in case the pipeline was removed
+        UpdatePasses(pipeline);
```

Comment:
This function is called when a RenderPipeline is removed too. 
you may want to handle that case